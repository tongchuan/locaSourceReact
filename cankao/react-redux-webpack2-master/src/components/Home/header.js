/**
 * Created by Administrator on 2016/7/1.
 */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export class Header extends Component {
    //注意，组件内部也可能有时候会暴露事件给外部调用，和参数传递的原理一样
    shouldComponentUpdate(nextProps, nextState) {
        if (_.isEqual(this.props, nextProps) || !_.isEmpty(this.props)) {
            return false
        }
        return true
    }

    render() {
        const {title, imgUrl, linkTo, bgColor, match} = this.props
        //提供4个接口参数给container做设置，可以不传参。
        console.log("match: ", match)
        return (
            <header className='header' style={bgColor}>
                {title}
                <Link to={`${match.url + linkTo}`} className="a_link" >
                    <img src={imgUrl} className="a_img" />
                </Link>
            </header>
        )
    }
    //严格来说，这些暴露给外部的参数都需要做验证,常用的验证类型为array，bool，func，number，object，string
    static propTypes = {
        title: React.PropTypes.string.isRequired
    }
}