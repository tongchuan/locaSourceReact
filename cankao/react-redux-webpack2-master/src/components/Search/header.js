/**
 * Created by Administrator on 2016/7/2.
 */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

const back = require('./files/back_search.png');
const search = require('./files/search_1.png');

//该组件没有做无状态优化处理，根据其他几个组件的优化方式，可以自行思考
export class Header extends Component {

    render() {
        return (
            <div className="style_body">
                <div className="style_main">
                    <Link to="/" className="style_a">
                        <img src={back} className="style_a_img" />
                    </Link>
                    <div className="style_div_content">
                        <img src={search} className="style_div_img" />
                        <input type="text" placeholder="搜索书名" className="style_div_input" />
                    </div>
                    <Link to="/search" className="style_right_a">搜索</Link>
                </div>
            </div>
        )
    }
}