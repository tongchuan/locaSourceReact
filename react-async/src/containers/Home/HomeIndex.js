import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import classNames from 'classnames'
import * as actions1 from '../../actions/user'
import * as actions2 from '../../actions/user0'
import {
  ButtonToolbar,
  ButtonGroup,
  Button,
  OverlayTrigger,
  Tooltip
 } from 'react-bootstrap';
import "./less/home.less"
/**
 * connect中间件
 * connect一定要写在需要传递参数的组件头部，因为这是语法规则，只对当前关联的组件生效，和java的原理是一致的
 * state用法：state => state（传递全部state） 或者 state => state.home（传递指定的state）
 * dispatch用法：（单个action）bindActionCreators(navActions, dispatch)，（多个action）bindActionCreators({...action1, ...action2}, dispatch)
 */

@connect(
    state => state,
    dispatch => bindActionCreators({...actions2, ...actions1}, dispatch)
)
export default class HomeIndex extends React.Component {
  constructor(props) {
    super(props);
    // console.log("home");
    console.log(this.props);
    this.updatename = this.updatename.bind(this)
  }
  componentDidMount(){
    console.log(this.props.sss())
  }

  updatename(){
    this.props.sss()
  }

  render() {
    // const tooltip = (
    //   <Tooltip id="tooltip"><strong>Holy guacamole!</strong> Check this info.</Tooltip>
    // );
    return (
      <div>
       <p>我是更新后的数据{this.props.user.name}</p>
        HomeIndex
        <button onClick={this.updatename}>点击我试一试更新数据状态</button>
      </div>
    )
  }
}
