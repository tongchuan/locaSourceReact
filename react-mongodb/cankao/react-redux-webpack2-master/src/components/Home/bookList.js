/**
 * Created by yongyuehuang on 2016/12/15.
 */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export class BookList extends Component {

    shouldComponentUpdate(nextProps, nextState) {
        if (_.isEqual(this.props, nextProps) || !_.isEmpty(this.props)) {
            return false
        }
        return true
    }

    render() {
        const {_id, imgUrl, title, author, press, publishedDate, currentPrice, originalPrice, index} = this.props
        return (
            <Link to={`/bookList/${_id}`} data-id={_id} className="home-book-list">
                <div className="book-img">
                    <span className="book-index">{index}</span>
                    <img className="" src={imgUrl} />
                </div>
                <div className="book-contain">
                    <p className="book-title textOverflow-hidden">{title}</p>
                    <p className="book-author textOverflow-hidden">作者：{author}</p>
                    <p className="book-author textOverflow-hidden">出版社：{press}</p>
                    {
                        publishedDate !== '' ?  <p className="book-author textOverflow-hidden">出版时间：{publishedDate}</p> : ''
                    }
                    <p className="textOverflow-hidden">
                        <span className="now-price">￥{currentPrice}</span>
                        <span className="old-price">￥{originalPrice}</span>
                    </p>
                </div>
            </Link>
        )
    }

    //数据验证
    static propTypes = {
        _id: React.PropTypes.string.isRequired,
        imgUrl: React.PropTypes.string.isRequired,
        title: React.PropTypes.string.isRequired,
        author: React.PropTypes.string.isRequired,
        press: React.PropTypes.string.isRequired,
        publishedDate: React.PropTypes.string.isRequired,
        currentPrice: React.PropTypes.string.isRequired,
        originalPrice: React.PropTypes.string.isRequired,
        index: React.PropTypes.number.isRequired //书籍序号
    }
}