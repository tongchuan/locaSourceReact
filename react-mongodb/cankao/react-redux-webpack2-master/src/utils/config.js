// 识别微信浏览器
export function isWeixinBrowser(){
    return /micromessenger/.test(navigator.userAgent.toLowerCase())
}