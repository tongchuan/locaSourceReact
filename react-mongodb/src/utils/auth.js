export function login(data){
  let token = Math.random().toString(36).substring(7)
  localStorage.token = token;
  localStorage.userName= data.email
  this.onChange(true);
}
export function getToken(){
  return localStorage.token;
}
export function logout(){
  delete localStorage.token;
  delete localStorage.userName;
  this.onChange(true);
}
export function loggedIn(){
  return !!localStorage.token
}
export function loggedName(){
  return !!localStorage.userName ? localStorage.userName : " "
}
export function onChange(){

}
