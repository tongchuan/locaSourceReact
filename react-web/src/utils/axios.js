import axios from 'axios'
import queryString from 'query-string'
import globalStore from '@/stores/common/globalStore'
import config from '@/stores/config'
// console.log(config)



let instance =  axios.create({
  baseURL:config.baseURL,
  method: 'get',
  headers: {
    'Content-Type': 'application/json;charset=utf-8'
  },
  // withCredentials: !!config.env,
  responseType: 'json',
  timeout: 6000,
  transformRequest: [
    function(data){
      return data
    }
  ],
  transformResponse:[function(data){
    // console.log(data)
    return data
  }]
})

// 添加请求拦截器
instance.interceptors.request.use(function (conf) {
  globalStore.showWait()
  // 在发送请求之前做些什么
  if(!config.env && conf.method.toLowerCase() === 'post'){
  //   // console.log(!config.env)
    conf.method = 'get'
  }

  if(conf.headers['Content-Type']!='application/json;charset=utf-8'){
    if(conf.method.toLowerCase() === 'post'){
      conf.data = queryString.stringify(conf.data)
    }else{
      conf.params = queryString.stringify(conf.data)
    }
  }else{
    if(conf.method.toLowerCase() === 'post'){
      conf.data = JSON.stringify(conf.data)
    }else{
      conf.params = JSON.stringify(conf.data)
    }
    // conf.data = JSON.stringify(conf.data)
    // conf.params = JSON.stringify(conf.data)
  }

  
  // if(conf.method.toLowerCase() === 'get' && conf.headers['Content-Type']!='application/json;charset=utf-8'){
  //   conf.params = JSON.parse(conf.data)
  // }
  // console.log(config)
  return conf;
}, function (error) {
  globalStore.hideWait()
  // 对请求错误做些什么
  return error;
});

// 添加响应拦截器
instance.interceptors.response.use(function (response) {
  globalStore.hideWait()
  // 对响应数据做点什么
  return response.data;
}, function (error) {
  globalStore.hideWait()
  globalStore.showError('数据请求失败,错误信息：'+error.toString())
  // 对响应错误做点什么
  return error;
});
export default instance