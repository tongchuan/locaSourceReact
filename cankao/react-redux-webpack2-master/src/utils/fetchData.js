/**
 * Created by admin on 2016/10/10.
 */
import axios from 'axios'

//封装好的get和post接口，调用方法情况action文件
export let instance = axios.create({
    // baseURL: BASE_URL, //设置默认api路径
    timeout: 10000, //设置超时时间
    headers: {'X-Custom-Header': 'foobar'}
});

export const getData = (url, param) => {
    return (
        instance.get(`${url}`, {
            params: param
        })
    )
}

export const postData = (url, param) => {
    return (
        instance.post(`${url}`, param)
    )
}