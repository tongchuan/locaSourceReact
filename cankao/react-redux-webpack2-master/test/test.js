/**
 * Created by yongyuehuang on 2017/3/28.
 */
import React from 'react';
import renderer from 'react-test-renderer';
const getBook = require('../src/actions/book');
import { getData, postData } from '../src/utils/fetchData';
import { Header } from '../src/components/Home/header';
const receiveHotSearch = require('../src/actions/search');

test('首页得到书籍列表', () => {
    expect(getBook).toThrowError();
});

test('async/await 函数测试', async () => {
    await expect(getData(`/api/book/list`)).resolve;
});

it('测试头部组件', () => {
    const tree = renderer.create(
        <Header title="测试头部" linkTo="http://www.baidu.com" />
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('测试搜索action === RECEIVE_HOT_SEARCH', () => {
    expect(receiveHotSearch)
})

test('adds 1 + 2 to equal 3', () => {
    expect(sum(1, 2)).toBe(3);
});

function sum(a, b) {
    return a + b;
}