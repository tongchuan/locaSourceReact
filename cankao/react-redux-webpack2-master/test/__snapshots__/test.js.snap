// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试头部组件 1`] = `
<header
  className="header"
  style={undefined}
>
  测试头部
  <a
    className="a_link"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      className="a_img"
      src={undefined}
    />
  </a>
</header>
`;
